crates/isograph_compiler/src/watch.rs

```diff
@@ -32,16 +37,23 @@ pub async fn handle_watch_command(
             .expect("Failing when watching schema extension");
     }
 
+    let mut state = CompilerState::new(config);
+    eprintln!("{}", "Starting to compile.".cyan());
+    let _ = print_result(WithDuration::new(|| state.compile()));
+
     tokio::spawn(async move {
         while let Some(res) = rx.recv().await {
             match res {
                 Ok(events) => {
-                    if any_modified_path_is_outside_artifact_directory(
-                        &events,
-                        &config.artifact_directory,
-                    ) {
-                        eprintln!("{}", "File changes detected.".cyan());
-                        let _ = compile_and_print(&config);
+                    if let Some(changes) = changed_files(&events, &state.config) {
+                        eprintln!("{}", "File changes detected. Starting to compile.".cyan());
+                        let _ = print_result(WithDuration::new(|| {
+                            if changes.len() < MAX_CHANGED_FILES {
+                                state.update(&changes)
+                            } else {
+                                state.compile()
```

Comment:
can we add a `eprintln` here indicating that we're falling back to a full compilation? Might be nice info for the user to see when they change branches